### DISCUSSION TITLE
Should wrapping comments be a job for the formatter?

### DISCUSSION THREAD

**Discussion by cyrus01337** [JohnnyMorganz/StyLua#874] [General]
Comments are useful for documentation or to explain niche situations - the latter being my common use case for them - and wondered if hard line-wrapping comments according to StyLua's line width property was ever a feature to be considered.

In VS Code there's already [an extension that handles this](https://marketplace.visualstudio.com/items?itemName=stkb.rewrap) that I use, though I'm wondering what tool should be responsible for this - the editor or extensions, and whether it can/should be handed over to formatters i.e StyLua?

**Comment by JohnnyMorganz** [comment]
Just noticed I missed this thread, sorry about that.

I've been thinking about comments wrapping before, and more generally just about overall formatting of comments.

My previous stance was that formatting comments was just not worth it, given that it could contain completely arbitrary text, and we might mess things up here.

But, I think a good balance might be to format documentation comments (https://github.com/JohnnyMorganz/StyLua/issues/851#issuecomment-1902069068).

I'm not sure if there is a unified consensus on what a documentation comment is though, but I think there are some general rules we could apply:

- Consider single line comments only, starting with `---`
- There must be a space between `---` and the text
- Consider general structure of annotations in doc comments when parsing (e.g., `@param`, `@returns` etc., and alignment with this - probably best to see how other formatting tools handle it for e.g. JSDoc)
- Enforce a line width for word wrapping (should it be the same as stylua's line width, or a separate configurable?)

All of this is assuming comments formatting would be easy to introduce, which I'm not sure about 😅. My main thing is whether we have enough context, or whether it would be invasive to introduce that context. I think it should be doable if we wanted to, the relevant code in `general.rs` around "load_token_trivia" would probably be sufficient. Also, some rules around word wrapping would need to be decided.

**Reply by cyrus01337** [reply]
I hadn't thought of the number of factors necessary to even consider this. It would be neat to wrap to the line width/column specified in the Stylua configuration file, however this is a lot to consider for what I initially thought was a simple implementation (suppose that's the world of formatters in general lol).

My only 2c is that having this opt-in would prevent invasive behavior and remain (maybe?) backwards-compatible (don't know the history of Stylua's changes from it's epoch).

What rules are missing from those you've listed exactly, just so we're on the same page?
